src/scores/spatial/fss_impl.py

```diff
@@ -0,0 +1,369 @@
+    da_fss = xr.apply_ufunc(
+        _fss_wrapper,
+        fcst,
+        obs,
+        input_core_dims=[list(spatial_dims), list(spatial_dims)],
+        vectorize=True,
+        dask=dask,  # pragma: no cover
+    )
+
+    # gather additional dimensions to aggregate over.
+    dims = gather_dimensions(
+        fcst.dims,
+        obs.dims,
+        reduce_dims=reduce_dims,
+        preserve_dims=preserve_dims,
+    )
+
+    if not (spatial_dims[0] in dims and spatial_dims[1] in dims):
+        # at least one spatial dim is trying to be preserved.
+        warnings.warn(
+            f"At least one of the provided spatial dims = {spatial_dims} is attempting "
+            "to be preserved. Please make sure `preserve_dims` and `reduce_dims` "
+            "are configured to not preserve `spatial_dims`, in order to suppress this "
+            "warning.",
+            DimensionWarning,
+        )
+
+    # trim any spatial dimensions as they shouldn't appear.
+    dims = set(dims) - set(spatial_dims)
+
+    # apply ufunc again but this time to compute the fss, reducing
+    # any non-spatial dimensions.
+    da_fss = xr.apply_ufunc(
+        _aggregate_fss_decomposed,
+        da_fss,
+        input_core_dims=[list(dims)],
+        vectorize=True,
+        dask=dask,  # pragma: no cover
+    )
+
+    return da_fss
+
+
+def fss_2d_binary(  # pylint: disable=too-many-locals,too-many-arguments
+    fcst: xr.DataArray,
+    obs: xr.DataArray,
+    *,  # Force keywords arguments to be keyword-only
+    window_size: Tuple[int, int],
+    spatial_dims: Tuple[str, str],
+    zero_padding: bool = False,
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    compute_method: FssComputeMethod = FssComputeMethod.NUMPY,
+    check_boolean: bool = True,
+    dask: str = "forbidden",  # see: `xarray.apply_ufunc` for options
+) -> XarrayLike:
+    """
+    Takes in a binary (True or False) field of fcst and obs events. For example
+    the output of a binary threshold applied to a continuous field or an event
+    operator.
+
+    Optionally the user can set ``check_boolean`` to ``True`` to check that the
+    field is boolean before any computations. Note: this asserts that the
+    underlying fields are of type ``np.bool_`` but does not coerce them. If
+    the user is confident that the input field is binary (but not necessarily
+    boolean), they may set this flag to ``False``, to allow for more flexible binary
+    configurations such as 0 and 1; this should give the same results.
+
+    Uses ``fss_2d`` from the ``scores.spatial`` module to perform fss
+    computation, but without the threshold operation.
+
+    .. seealso::
+        :py:func:`scores.spatial.fss_2d` for more details and the continuous
+        version. As well as detailed argument definitions.
+    """
+
+    if check_boolean:
+        if not (fcst.dtype == np.bool_ and obs.dtype == np.bool_):
```

Comment:
What if there is a NaN in there?

Reply:
Then it shouldn't be valid, it's underlying type would probably be float. `check_boolean` is supposed to be a rigid condition. Handling "empty" data is actually not trivial, and is beyond the scope of this function.